pkl-core/src/main/java/org/pkl/core/ast/type/ResolveDeclaredTypeNode.java

```diff
@@ -73,11 +74,36 @@ protected VmTyped getImport(
     var result = module.getCachedValue(importName);
     if (result == null) {
       result = callNode.call(member.getCallTarget(), module, module, importName);
+
+      var importedModule = (VmTyped) result;
+      if (importedModule.isNotInitialized()
+          && importedModule.isModuleObject()
+          && importedModule.getModuleInfo().isAmend()) {
+        // this is an amending module. Try to find the prototype
+        var prototypeModule = findPrototypeModule(importedModule);
+        if (prototypeModule != null) {
+          result = prototypeModule;
+        }
+      }
+
       module.setCachedValue(importName, result);
```

Comment:
We shouldn't be cacheing this under the same name as the import value. Otherwise, this snippet breaks:

```pkl
import "amendingFoo.pkl"

// here, `amendingFoo` should resolve to the prototype
res: amendingFoo.Foo

// here, `amendingFoo` should resolve to the import itself
bar = amendingFoo
```